### ISSUE TITLE
IndexError: list index out of range when connecting to OpenLDAP server

### ISSUE THREAD

**Issue by deb761** [cannatag/ldap3#368]
The connection works with ldap3 1.0.3, but fails on 2.2.4.

Version 2.4.41 of openldap

ldap3 1.0.3 packages:
```
click==6.7
Flask==0.12.2
flask-ldap3-login==0.9.12
Flask-WTF==0.14.2
itsdangerous==0.24
Jinja2==2.9.6
ldap3==1.0.3
MarkupSafe==1.0
pyasn1==0.2.3
Werkzeug==0.12.2
WTForms==2.1
```
ldap3 2.2.4 packages:
```
click==6.7
Flask==0.12.2
flask-ldap3-login==0.9.12
Flask-WTF==0.14.2
itsdangerous==0.24
Jinja2==2.9.6
ldap3==2.2.4
MarkupSafe==1.0
pyasn1==0.2.3
Werkzeug==0.12.2
WTForms==2.1
```

code used for connection:
```
"""Test connecting to ldap3"""

from ldap3 import Connection

uid = '<uid>'
password = '<password>'

conn = Connection("ldap.nsd.org",
     user='uid=ldapauth,ou=people,dc=nsd,dc=org',password='<ldapauth-password>',
     auto_bind=True)
conn.start_tls()

conn.search('ou=people,dc=nsd,dc=org',
            '(&(uid={})(objectclass=nsdstaff))'.format(uid))
if conn.entries:
    user = conn.entries[0]
    print(dir(user))
    if Connection("ldap.nsd.org", user.entry_get_dn(), password).bind():
        print(uid)
    else:
        print("Uid error")
else:
    print("Not a staff member")

conn.unbind()
```
Error:
```
Traceback (most recent call last):
  File "test_ldap.py", line 10, in <module>
    auto_bind=True)
  File "/Users/dengelmeyer/Development/flaskApp/env/lib/python3.6/site-packages/ldap3/core/connection.py", line 328, in __init__
    self.bind(read_server_info=True)
  File "/Users/dengelmeyer/Development/flaskApp/env/lib/python3.6/site-packages/ldap3/core/connection.py", line 595, in bind
    self.refresh_server_info()
  File "/Users/dengelmeyer/Development/flaskApp/env/lib/python3.6/site-packages/ldap3/core/connection.py", line 1315, in refresh_server_info
    self.server.get_info_from_server(self)
  File "/Users/dengelmeyer/Development/flaskApp/env/lib/python3.6/site-packages/ldap3/core/server.py", line 441, in get_info_from_server
    self._get_schema_info(connection)
  File "/Users/dengelmeyer/Development/flaskApp/env/lib/python3.6/site-packages/ldap3/core/server.py", line 388, in _get_schema_info
    schema_entry = connection.response[0]['raw_attributes']['subschemaSubentry'][0]
IndexError: list index out of range
```

**Comment by cannatag** [comment]
It seems that your server is not returning he schema.

Try to define the Server object with get_info=NONE.

Also I don't understand why you need a new Connection in the ``if Connection(...`` statement. Can't you rebound on the same connection?

**Comment by cannatag** [comment]
Please check the code in dev. I've added an additional checking that should avoid the error even with default get_info parameter.

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.
